emperor/support_files/js/color-view-controller.js

```diff
@@ -0,0 +1,214 @@
+    // the second array has the identifiers used to generate the colors
+    value = ColorViewController.Colormaps[i];
+    name = ColorViewController.ColormapNames[i];
+    colorItem = $('<option>').attr('value', value).text(name);
+
+    scope.$colormapSelect.append(colorItem);
+  }
+
+  // Build the options dictionary
+  var options = {'valueUpdatedCallback':function(e, args) {
+                   var val = args.item.category, color = args.item.color, group = [];
+                   group = args.item.plottables;
+                   scope.decompViewDict[scope.getActiveDecompViewKey()].setGroupColor(color, group);
+                 },
+                 'categorySelectionCallback':function(evt, params) {
+                   // we re-use this same callback regardless of whether the
+                   // color or the metadata category changed, maybe we can do
+                   // something better about this
+                   var category = scope.$select.val();
+                   var color = scope.$colormapSelect.val();
+
+                   var k = scope.getActiveDecompViewKey();
+                   var data;
+
+                   // fetch the slickgrid-formatted data
+                   data = scope.decompViewDict[k].setCategoryColors(color,
+                                                                    category);
+
+                   scope.setSlickGridDataset(data);
+                 },
+                 'slickGridColumn':{id: 'title', name: '', field: 'color',
+                                    sortable: false, maxWidth: SLICK_WIDTH,
+                                    minWidth: SLICK_WIDTH,
+                                    editor: ColorEditor,
+                                    formatter: ColorFormatter}};
+
+  EmperorAttributeABC.call(this, container, title, helpmenu,
+                           decompViewDict, options);
+  this.$header.append(this.$colormapSelect);
+
+  // the chosen select can only be set when the document is ready
+  $(function() {
+    scope.$colormapSelect.chosen({width: "100%", search_contains: true});
+    scope.$colormapSelect.chosen().change(options.categorySelectionCallback);
+  });
+
+  return this;
+}
+ColorViewController.prototype = Object.create(EmperorAttributeABC.prototype);
+ColorViewController.prototype.constructor = EmperorAttributeABC;
+
+
+/**
+ *
+ * Generate a list of colors that corresponds to all the samples in the plot
+ *
+ * @param {values} list of objects to generate a color for, usually a category
+ * in a given metadata column.
+ * @param {map} name of the color map to use, see ColorViewController.Colormaps
+ *
+ *
+ * This function will generate a list of coloring values depending on the
+ * coloring scheme that the system is currently using (discrete or continuous).
+*/
+ColorViewController.getColorList = function(values, map) {
+  var colors = {}, numColors = values.length-1, counter=0, interpolator,
```

Comment:
Not sure if we want to have in separate lines those vars that have an initial value and those that don't.